homeassistant/components/mystrom/switch.py

```diff
@@ -101,3 +113,28 @@ async def async_update(self):
             if self._available:
                 self._available = False
                 _LOGGER.error("No route to myStrom plug")
+
+
+class MyStromPowerSensor(SensorEntity):
```

Comment:
You can't add a sensor entity from a switch platform, it has to happen from a `sensor.py`

Reply:
good to know. So basically there is no relation between the sensor and the switch? they are completely independent entities? also the data fetching instance can't be shared or is there some kind of dependency injection?

Reply:
It seems this integration uses a single call to fetch all data. In those cases, the `DataUpdateCoordinator` can be used to handle the polling centrally.

https://developers.home-assistant.io/docs/integration_fetching_data#coordinated-single-api-poll-for-data-for-all-entities

The coordinator will get the state on an interval, and notify update all entities (on all platforms) when it got the new state.

Reply:
I‘ll do the coordinator improvements in a follow up PR. This PR should now be ready.

Reply:
The problem with adding the additional sensors is its now going to poll the device additional times to update each sensor.

I'm not sure we allow adding new platforms without refactoring for https://github.com/home-assistant/architecture/blob/master/adr/0007-integration-config-yaml-structure.md first

It would probably make more sense to do convert to a config flow though https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md

Reply:
Yeah, this conflicts with ADR-0007.

> It would probably make more sense to do convert to a config flow though

Agreed

Reply:
I tried to add a config flow. But I'm not sure what is really necessary to have an integration to actually show up. Setting the config flow flag in the manifest and adding a basic config flow file didn't work. The documentation and example repo is quite lacking.

do you have some pointers to understand what is really required to get a basic config flow up and running?

Reply:
I was reading the [Config Flow](https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md) documentation and I don't think that applies to the MyStrom switches. 
In particular:

> However, in the first two cases, it doesn’t work. Integrations are discovered. Integrations require users logging in on the vendor’s website and authorize linking (OAuth2) or users are required to press buttons on the hub to authorize linking (i.e. Hue).

MyStrom switches are accessed by IP-Address and REST-API as long as their reachable via network. Its not needed to login somewhere or do something on the device for it to show up somewhere. So, no discovery. 

The easiest way to configure these would be to add a IP and name for every switch/sensor added. Can this be done via  [Config Flow](https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md) ?


Regarding Refactoring for [YAML Structure](https://github.com/home-assistant/architecture/blob/master/adr/0007-integration-config-yaml-structure.md)

This would look like this:
```
mystrom:
   mystrom_dev_1:
     ip: 192.168.0.2
     name: mystrom_ff
   mystrom_dev_2:
     ip: 192.168.0.3
     name: mystrom_ff
```
And that would add entities like the switches and also power sensors eg. `sensor.mystrom_dev_1_power`

Correct?

Is there a sample which can be roughly followed?

Reply:
`python3 -m script.scaffold config_flow` can help you get started with a config flow

Reply:
I gave it a try and added the config flow in PR 74719.